### PR TITLE
cmd/snap,image: initial support for Core 20 in prepare-image with test

### PR DESCRIPTION
**Pull request by pedronis**
MakeBootable and its use will need adjustment after the bootenv handling is clarified/implemented.

Had to change the interface between cmd/snap/cmd_prepare_image.go and image.go.

### CODE REVIEW

**Comment by pedronis** [comment]
based on #7694

File: image/image.go [hunk 1]
```diff
@@ -168,17 +165,44 @@ func MockTrusted(mockTrusted []asserts.Assertion) (restore func()) {
 	}
 }
 
+func makeLabel(now time.Time) string {
+	return now.UTC().Format("20060102")
```
Comment (mvo5) [inline image/image.go]: Nice

File: image/image.go [hunk 2]
```diff
@@ -168,17 +165,44 @@ func MockTrusted(mockTrusted []asserts.Assertion) (restore func()) {
...
-	if snaps, _ := filepath.Glob(filepath.Join(dirs.SnapBlobDirUnder(opts.RootDir), "*.snap")); len(snaps) > 0 {
-		return fmt.Errorf("need an empty snap dir in rootdir, got: %v", snaps)
+	core20 := model.Grade() != asserts.ModelGradeUnset
+	var rootDir string
+	var bootRootDir string
+	var seedDir string
+	var label string
+	if !core20 {
+		// Core 16/28,  writing for the writeable partion
+		rootDir = filepath.Join(opts.PrepareDir, "image")
```
Comment (pedronis) [inline image/image.go]: there's a bug here, we shouldn't add `image` for classic

File: image/image.go [hunk 3]
```diff
@@ -168,17 +165,44 @@ func MockTrusted(mockTrusted []asserts.Assertion) (restore func()) {
...
-	}
-	if snaps, _ := filepath.Glob(filepath.Join(dirs.SnapBlobDirUnder(opts.RootDir), "*.snap")); len(snaps) > 0 {
-		return fmt.Errorf("need an empty snap dir in rootdir, got: %v", snaps)
+	core20 := model.Grade() != asserts.ModelGradeUnset
+	var rootDir string
+	var bootRootDir string
+	var seedDir string
+	var label string
+	if !core20 {
+		// Core 16/28,  writing for the writeable partion
```
Comment (anonymouse64) [inline image/image.go]: typo 
```suggestion
		// Core 16/18,  writing for the writeable partion
```

unless we have already a UC28?!?!?!

File: image/image.go [hunk 4]
```diff
@@ -380,17 +407,22 @@ func setupSeed(tsto *ToolingStore, model *asserts.Model, opts *Options) error {
...
-		return err
+	// cloud-init config (done at install for Core 20)
+	if !core20 {
+		// and the cloud-init things
+		if err := installCloudConfig(rootDir, gadgetUnpackDir); err != nil {
+			return err
+		}
 	}
 
 	return nil
```
Comment (anonymouse64) [inline image/image.go]: reading this function start to end makes me feel like it needs to be split up... not for this PR, but a 250 line long function is not a fun thing to debug and track :confused:

File: image/image.go (hunk 3 above)
Comment (chipaca) [inline image/image.go]: 28? :-)

File: image/image.go (hunk 2 above)
Comment (anonymouse64) [reply image/image.go]: do we need to make a modification to TestSetupSeedClassic to account for this?

**Comment by anonymouse64** [comment]
looks like the spread tests failed from some store 403s :-/

**Pull request merged**